lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -124,12 +132,19 @@ def visit_call_node(node)
     def visit_def_node(node)
       method_name = node.name.to_s
       comments = collect_comments(node)
-      case node.receiver
+      entry_class = case node.receiver
       when nil
-        @index << Entry::InstanceMethod.new(method_name, @file_path, node.location, comments, node.parameters)
+        if @singleton_class # i.e. `class << self`
+          Entry::SingletonMethod
+        else
+          Entry::InstanceMethod
+        end
       when Prism::SelfNode
-        @index << Entry::SingletonMethod.new(method_name, @file_path, node.location, comments, node.parameters)
+        Entry::SingletonMethod
+      else
```

Comment:
We don't yet handle things like `def Foo.bar`. The next step will be record the namespace along with the definition.